app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -241,25 +245,38 @@ def build_metric_id(type, resource, metric_id)
     # by ems_ref, which in Hawkular terms is the
     # fully qualified resource path from Hawkular inventory
     def run_generic_operation(operation, ems_ref, parameters = {})
-      with_provider_connection do |connection|
         the_operation = {
           :operationName => operation,
           :resourcePath  => ems_ref.to_s,
           :parameters    => parameters
         }
+        run_operation(the_operation)
+    end
 
+    def run_specific_operation(operation_name, ems_ref, parameters = {})
+      parameters[:resourcePath] = ems_ref.to_s
+      run_operation(parameters, operation_name)
+    end
+
+    def run_operation(parameters, operation_name = nil)
+      with_provider_connection do |connection|
         actual_data = {}
```

Comment:
I was wondering the same..  and from what I can see there is no point on have `actual_data` variable, at least in the case of this method, because it doesn't use the response. I'll remove it.